core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/service/asset/AssetServiceImpl.java

```diff
@@ -109,4 +109,32 @@ public ServiceResult<Asset> delete(String assetId) {
             return ServiceResult.success(deleted);
         });
     }
+
+    @Override
+    public ServiceResult<Void> update(String assetId, Asset asset) {
+        return transactionContext.execute(() -> {
+            if (findById(assetId) == null) {
+                return ServiceResult.notFound(format("Asset %s cannot be updated because it does not exist", assetId));
+            }
+            var updatedAsset = index.updateAsset(assetId, asset);
+            observable.invokeForEach(l -> l.updated(updatedAsset));
+
+            return ServiceResult.success(null);
```

Comment:
nit: we could add a `success()` method on `ServiceResult` like it was done on `Result` to hide the `null` thing

Reply:
good point. will do